finesse/hardware/plugins/decades/decades.py

```diff
@@ -120,9 +65,91 @@ def __init__(
             poll_interval: How often to poll the sensors (seconds)
         """
         super().__init__(DECADES_URL.format(host=host))
+        self._poll_interval = poll_interval
         self._poll_timer = QTimer()
         self._poll_timer.timeout.connect(self.send_data)
-        self._poll_timer.start(int(poll_interval * 1000))
+        self._poll_timer.start(int(self._poll_interval * 1000))
+
+        # Obtain full parameter list in order to parse received data
+        self.send_params()
+
+        # Poll device once on open.
+        # TODO: Run this synchronously so we can check that things work before the
+        # device.opened message is sent
+        self.send_data()
+
+    def send_params(self) -> None:
+        """Request the parameter list from the DECADES server.
+
+        The HTTP request is made on a background thread.
+        """
+        self._requester.make_request(
+            self._url + "/params",
+            self.pubsub_broadcast(self._on_params_received, "params.response"),
+        )
+
+    def send_data(self) -> None:
```

Comment:
And this `get_data`?

Reply:
Ditto above. Let's rename it when we port to `SensorsBase`